kernel/idle.c

```diff
@@ -27,6 +28,11 @@ void idle(void *unused1, void *unused2, void *unused3)
 	__ASSERT_NO_MSG(arch_current_thread()->base.prio >= 0);
 
 	while (true) {
+#ifdef CONFIG_IDLE_ENTER_HOOK
+		idle_enter_hook();
+#endif /* CONFIG_IDLE_ENTER_HOOK */
+
+#ifdef CONFIG_IDLE_DEFAULT_ROUTINE
```

Comment:
Point of clarification ... Does/should the setting of IDLE_ENTER_HOOK influence the setting of IDLE_DEFAULT_ROUTINE (or vice versa)? I ask, because I note that if both are set to 'n', then we get an idle loop that is essentially while (1)--not inherently wrong, but it certainly raises an eyebrow.

Reply:
The "true" default idle loop would just be to spin the CPU in a while(1), that is, if idle logic has not been implemented for an SoC using the hook, don't idle.

Optimistically the `IDLE_DEFAULT_ROUTINE` will be deprecated, as SoC specific hooks are implemented over time. So the only options will be implement hook or spin :)

Various default routines can still be written and used, like one which just calls `k_cpu_idle()`, they would just be selected by calling them from the hook :)
```
void idle_enter_hook(void)
{
        idle_hook_default_cpu_idle();
}
```

Reply:
Looking at this, maybe the default routine should actually just be an implementation of the `idle_enter_hook()` which selects `IDLE_ENTER_HOOK`